jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -290,147 +290,225 @@ ecma_builtin_global_object_parse_float (ecma_value_t this_arg __attr_unused___,
+    JERRY_ASSERT (bytes_copied >= 0);
+    lit_utf8_iterator_t iter = lit_utf8_iterator_create (utf8_string_buff, str_size);
 
-  /* Check if sign is present. */
-  if (utf8_string_buff[start] == '-')
-  {
-    sign = true;
-    start++;
-  }
-  else if (utf8_string_buff[start] == '+')
-  {
-    start++;
-  }
+    lit_utf8_iterator_seek_eos (&iter);
 
-  ecma_number_t *ret_num_p = ecma_alloc_number ();
+    lit_utf8_iterator_pos_t start = lit_utf8_iterator_get_pos (&iter);
+    lit_utf8_iterator_pos_t end = lit_utf8_iterator_get_pos (&iter);
 
-  /* Check if string is equal to "Infinity". */
-  const lit_utf8_byte_t *infinity_utf8_str_p = lit_get_magic_string_utf8 (LIT_MAGIC_STRING_INFINITY_UL);
+    lit_utf8_iterator_seek_bos (&iter);
 
-  for (lit_utf8_size_t i = 0; infinity_utf8_str_p[i] == utf8_string_buff[start + i]; i++)
-  {
-    if (infinity_utf8_str_p[i + 1] == 0)
+
+    /* 2. Find first non whitespace char and set starting position. */
+    while (!lit_utf8_iterator_is_eos (&iter))
     {
-      *ret_num_p = ecma_number_make_infinity (sign);
-      ret_value = ecma_make_normal_completion_value (ecma_make_number_value (ret_num_p));
-      break;
+      ecma_char_t current_char = lit_utf8_iterator_read_next (&iter);
+
+      if (!lit_char_is_white_space (current_char)
+          && !lit_char_is_line_terminator (current_char))
+      {
+        lit_utf8_iterator_read_prev (&iter);
+        start = lit_utf8_iterator_get_pos (&iter);
+        break;
+      }
     }
-  }
 
-  if (ecma_is_completion_value_empty (ret_value))
-  {
-    lit_utf8_size_t current = start;
-    lit_utf8_size_t end = str_size;
-    bool has_whole_part = false;
-    bool has_fraction_part = false;
+    bool sign = false;
 
-    if (lit_char_is_decimal_digit (utf8_string_buff[current]))
+    /* Check if sign is present. */
+    ecma_char_t current = lit_utf8_iterator_read_next (&iter);
+    if (current == LIT_CHAR_MINUS)
     {
-      has_whole_part = true;
+      sign = true;
+    }
 
-      /* Check digits of whole part. */
-      for (lit_utf8_size_t i = current; i < str_size; i++, current++)
+    if (current == LIT_CHAR_MINUS || current == LIT_CHAR_PLUS)
+    {
+      /* Set starting position to be after the sign character. */
+      start = lit_utf8_iterator_get_pos (&iter);
+    }
+    else
+    {
+      lit_utf8_iterator_read_prev (&iter);
+    }
+
+    ecma_number_t *ret_num_p = ecma_alloc_number ();
+
+    const lit_utf8_byte_t *infinity_utf8_str_p = lit_get_magic_string_utf8 (LIT_MAGIC_STRING_INFINITY_UL);
+    lit_utf8_iterator_t infinity_iter = lit_utf8_iterator_create (infinity_utf8_str_p,
+                                                                  sizeof (*infinity_utf8_str_p));
+
+    JERRY_ASSERT (!lit_utf8_iterator_is_eos (&infinity_iter));
+
+    /* Check if string is equal to "Infinity". */
+    while (!lit_utf8_iterator_is_eos (&iter)
+           && (lit_utf8_iterator_read_next (&iter) == lit_utf8_iterator_read_next (&infinity_iter)))
+    {
```

Comment:
We can't use `lit_compare_utf8_string_and_magic_string` here, because we have check if a prefix of the string contains "Infinity". For example, "Infinityfoo" should also be valid.